chips/cc26x2/src/commands.rs

```diff
@@ -0,0 +1,314 @@
+#![allow(dead_code)]
+use kernel::common::regs::ReadOnly;
+
+/******************************************************************************
+ * Radio and data commands bitfields
+*******************************************************************************/
+
+bitfield! {
+    #[derive(Copy, Clone)]
+    pub struct RfcTrigger(u8);
+    impl Debug;
+    pub _trigger_type, _set_trigger_type : 3, 0;
+    pub _enable_cmd, _set_enable_cmd      : 4;
+    pub _trigger_no, _set_trigger_no      : 6, 5;
+    pub _past_trigger, _set_past_trigger  : 7;
+
+}
+
+bitfield! {
+    #[derive(Copy, Clone)]
+    pub struct RfcCondition(u8);
+    impl Debug;
+    pub _rule, set_rule : 3, 0;
+    pub _skip, _set_skip : 7, 4;
+}
+
+bitfield! {
+    #[derive(Copy, Clone)]
+    pub struct RfcSetupConfig(u16);
+    impl Debug;
+    pub _frontend_mode, set_frontend_mode: 2, 0;
+    pub _bias_mode, set_bias_mode: 3;
+    pub _analog_cfg_mode, _set_analog_config_mode: 9, 4;
+    pub _no_fs_powerup, _set_no_fs_powerup: 10;
+}
+/******************************************************************************
+ * Radio Command structure headers, bitfields, and partial settings for the
+ * bitfields
+ *
+*******************************************************************************/
+
+#[derive(Clone, Copy)]
+pub enum TriggerType {
+    Now = 0,
+    Never = 1,
+    AbsTime = 2,
+    Submit = 3,
+    RelStart = 4,
+    RelPrevStart = 5,
+    RelFirstStart = 6,
+    RelPrevEnd = 7,
+    RelEvt1 = 8,
+    RelEvt2 = 9,
+    External = 10,
+}
+
+#[derive(Clone, Copy)]
+pub enum ConditionRules {
+    Always = 0,
+    Never = 1,
+    StopOnFalse = 2,
+    StopOnTrue = 3,
+    SkipOnFalse = 4,
+    SkipOnTrue = 5,
+}
+
+pub struct DataEntryQueue {
+    p_curr_entry: *mut u32,
+    p_last_entry: *mut u32,
+}
+
+/******************************************************************************
+ * Radio Commands
+ *
+*******************************************************************************/
+/*
+    RFC Immediate commands
+*/
+pub const RFC_CMD0: u16 = 0x607;
+pub const RFC_PING: u16 = 0x406;
+pub const RFC_BUS_REQUEST: u16 = 0x40E;
+pub const RFC_START_RAT_TIMER: u16 = 0x080A;
+pub const RFC_STOP_RAT_TIMER: u16 = 0x0809;
+pub const RFC_SETUP: u16 = 0x0802;
+pub const RFC_STOP: u16 = 0x0402;
+pub const RFC_FS_POWERDOWN: u16 = 0x080D;
+
+pub struct DirectCommand {
```

Comment:
Since this is just dictating structure, and it's the size of single register, I think it would be useful to make this `#[derive(Clone,Copy)]` (see comment on `send_direct`

Reply:
added